### PR TITLE
Client label node

### PR DESCRIPTION
**Pull request by squall0gd**
This is proposal of node labeling implementation from client side. Second approach is  [kubelet side](#41)

### CODE REVIEW

File: cluster/addons/iso-client/coreaffinity-isolator.go [hunk 1]
```diff
@@ -45,6 +46,11 @@ func shutdownIsolator(client *aff.EventDispatcherClient, opaque *opaq.OpaqueInte
 		glog.Fatalf("Failed to remove opaque resources: %v", err)
 	}
 
+	glog.Info("Removing label from node")
+	if err := utils.UnsetLabel(name); err != nil {
+		glog.Fatalf("cannot unset label from node: %q", err)
```
Comment (flyingcougar) [inline cluster/addons/iso-client/coreaffinity-isolator.go]: "Failed to"

File: cluster/addons/iso-client/coreaffinity-isolator.go [hunk 2]
```diff
@@ -78,6 +84,12 @@ func main() {
 		glog.Errorf("Cannot register isolator: %v", err)
 		shutdownIsolator(nil, opaque)
 	}
+
+	glog.Infof("Mark node that isolator %q is available", name)
+	if err := utils.SetLabel(name); err != nil {
+		glog.Errorf("cannot mark node: %q", err.Error())
```
Comment (flyingcougar) [inline cluster/addons/iso-client/coreaffinity-isolator.go]: "Failed to label node ...."

File: cluster/addons/iso-client/coreaffinity-isolator.go [hunk 3]
```diff
@@ -78,6 +84,12 @@ func main() {
 		glog.Errorf("Cannot register isolator: %v", err)
 		shutdownIsolator(nil, opaque)
 	}
+
+	glog.Infof("Mark node that isolator %q is available", name)
```
Comment (flyingcougar) [inline cluster/addons/iso-client/coreaffinity-isolator.go]: s/Mark/Label

File: cluster/addons/iso-client/utils/labels.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	} else {
+		delete(labels, isolatorName)
+	}
+
+	node.SetLabels(labels)
+	_, err = client.Nodes().Update(node)
+	return err
+}
+
+func SetLabel(isolatorName string) error {
```
Comment (flyingcougar) [inline cluster/addons/iso-client/utils/labels.go:50]: very user friendly wrap :+1:

File: cluster/addons/iso-client/utils/labels.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+)
+
+func generateIsolatorName(name string) string {
+	return fmt.Sprintf("external-isolator-%s", name)
+}
+
+func applyLabelChange(isolatorName string, add bool) error {
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return fmt.Errorf("cannot get in-cluster config: %q", err)
```
Comment (flyingcougar) [inline cluster/addons/iso-client/utils/labels.go]: "Failed to get in-cluster config"

File: cluster/addons/iso-client/utils/labels.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	nodename, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("cannot gather hostname: %q", err)
```
Comment (flyingcougar) [inline cluster/addons/iso-client/utils/labels.go]: "Failed to read hostname"

File: cluster/addons/iso-client/utils/labels.go [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	clientset, err := kubeapi.NewForConfig(config)
+	if err != nil {
+		return fmt.Errorf("cannot achieve client instance for Kubernetes: %s", err)
```
Comment (flyingcougar) [inline cluster/addons/iso-client/utils/labels.go]: "Failed to connect initialize client connection" or sth simmilar

File: cluster/addons/iso-client/utils/labels.go [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	client := clientset.CoreV1()
+	node, err := client.Nodes().Get(nodename, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("cannot get node %q: %q", nodename, err)
```
Comment (flyingcougar) [inline cluster/addons/iso-client/utils/labels.go]: "Failed to"....

File: cluster/addons/iso-client/utils/labels.go [hunk 9]
```diff
@@ -0,0 +1,56 @@
...
+	}
+
+	labels := node.GetLabels()
+	if add {
+		labels[isolatorName] = ""
+	} else {
+		delete(labels, isolatorName)
```
Comment (flyingcougar) [inline cluster/addons/iso-client/utils/labels.go:42]: what will happen if the label has meed deleted manually by user and then `delete(labels, isolatorName)` is called ? error, exception ?

**Review by flyingcougar — CHANGES_REQUESTED** [review]

File: cluster/addons/iso-client/utils/labels.go (hunk 9 above)
Comment (squall0gd) [reply cluster/addons/iso-client/utils/labels.go:42]: https://golang.org/pkg/builtin/#delete

**Comment by ConnorDoyle** [comment]
Are we dropping this PR in favor of #41?

**Comment by flyingcougar** [comment]
@ConnorDoyle - i'd agree that we should drop this in favor of #41

**Pull request closed without merging**